File: tests/loop/split_range/loop_split_range_doc.ml
open Optitrust
open Prelude

let _ = Flags.check_validity := true

let _ = Run.script_cpp (fun _ ->

  !! Loop_basic.split_range ~nb:2 [cFor "i"];
     Loop_basic.split_range ~cut:(expr "cut") [cFor "k"];

  !!! (); (* TODO: Find how to eliminate this reparse *)
)

File: tests/loop/split_range/loop_split_range_doc.cpp
#include <optitrust.h>

void f() {
  __pure();

  int x = 0;
  for (int i = 0; i < 10; i++){
    x += i;
  }
  const int cut = 2;
  for (int k = 0; k < 10; k++){
    x += k;
  }

}
